tests/nodetree/test_node_unit.py

```diff
@@ -748,14 +780,17 @@ async def test_wait_for_state_change(
         expect_early_termination,
     ):
         node = MockLeafNode(())
-        node.subscribe = Mock(return_value=self._get_future("queue"))
-        future = self._get_future(AnnotatedValue(path="path", value=current_value))
+        node.subscribe = Mock(return_value=_get_future("queue"))
+        future = _get_future(AnnotatedValue(path="path", value=current_value))
 
         with patch(
-            "labone.nodetree.node.LeafNode._wait_for_state_change_loop",
+            "labone.nodetree.node.LeafNode._wait_for_state_change_loop", autospec=True,
         ) as loop_patch, patch(
             "labone.nodetree.node.Node.__call__",
             MagicMock(return_value=future),
+            # TODO: adding autospec=True breaks this because we already give in a mock.
```

Comment:
if you already pass in a Mock you do not need patch but can make __call__ of the Mock directly return a the future ... or what am I missing? 

In any case, when mocking a magic function I would say autospec is neglectable since this interface will not change (unless python itself does a major breaking change